clients/js/src/embeddings/OpenAIEmbeddingFunction.ts

```diff
@@ -1,45 +1,123 @@
-import { IEmbeddingFunction } from "./IEmbeddingFunction";
+import {IEmbeddingFunction} from "./IEmbeddingFunction";
+
 let OpenAIApi: any;
+let openAiVersion = null;
+let openAiMajorVersion = null;
+
+// interface EmbeddingResponse {
+//     object: string;
+//     data: { embedding: number[] }[];
+//     model: string;
+//     usage: { prompt_tokens: number; total_tokens: number };
+// }
+
+interface OpenAIAPI {
+    createEmbedding: (params: {
+        model: string;
+        input: string[];
+        user?: string;
+    }) => Promise<object>;
+}
+
+class OpenAIAPIv3 implements OpenAIAPI {
+    private readonly configuration: any;
+    private openai: any;
+
+    constructor(configuration: { organization: string, apiKey: string }) {
+        this.configuration = new OpenAIApi.Configuration({
+            organization: configuration.organization,
+            apiKey: configuration.apiKey,
+        });
+        this.openai = new OpenAIApi.OpenAIApi(this.configuration);
+    }
+
+    public async createEmbedding(params: {
+        model: string,
+        input: string[],
+        user?: string
+    }): Promise<object> {
+        const response = await this.openai.createEmbedding({
+            model: params.model,
+            input: params.input,
+        });
+        return response.data;
+    }
+}
+
+class OpenAIAPIv4 implements OpenAIAPI {
+    private readonly apiKey: any;
+    private openai: any;
+
+    constructor(apiKey: any) {
+        this.apiKey = apiKey;
+        this.openai = new OpenAIApi({
+            apiKey: this.apiKey,
+        });
+    }
+
+    public async createEmbedding(params: {
+        model: string,
+        input: string[],
+        user?: string
+    }): Promise<object> {
+        return await this.openai.embeddings.create(params);
+    }
+}
 
 export class OpenAIEmbeddingFunction implements IEmbeddingFunction {
     private api_key: string;
     private org_id: string;
     private model: string;
+    private openaiApi: OpenAIAPI;
 
     constructor({
-        openai_api_key,
-        openai_model,
-        openai_organization_id,
-    }: {
+                    openai_api_key,
+                    openai_model,
+                    openai_organization_id,
+                }: {
```

Comment:
remove these indents?

Reply:
Tabs vs Spaces 🤣 . Using PyCharm for most of my coding, not optimal for JS/TS.